subprojects/launcher/src/main/java/org/gradle/launcher/cli/converter/LayoutToPropertiesConverter.java

```diff
@@ -75,8 +78,14 @@ private void maybeConfigureFrom(File propertiesFile, Map<String, String> result)
             throw new UncheckedIOException(e);
         }
 
-        for (Object key : properties.keySet()) {
-            if (GradleProperties.ALL.contains(key.toString())) {
+        for (final Object key : properties.keySet()) {
+            GradleBuildOption validOption = CollectionUtils.findFirst(GradleBuildOptions.ALL, new Spec<GradleBuildOption>() {
```

Comment:
⭕️  I feel like it would be a bit easier to grok this code if we did something like `GradleBuildOptions.ALL.map(it.getGradleProperty()).contains(key.toString())`